### DISCUSSION TITLE
Using tools with cake.frosting

### DISCUSSION THREAD

**Discussion by nathangiuliani** [cake-build/cake#2931] [Q&A] [answered]
Hi all,

I'm looking at switching a number of our build projects to Cake Frosting, because I find the experience of editing with an IDE that's working to be far easier :)

However I can't quite figure out how to use tools such as those referenced in these lines:

```
#tool "nuget:?package=OctopusTools&version=7.4.2"
#tool "nuget:?package=NUnit.ConsoleRunner&version=3.11.1"
#tool "nuget:?package=OpenCover&version=4.7.922"
```

I have things working by including the nugets, copying the tools as part of msbuild and registering the executables "manually", but I would have thought there was a better solution.

Thanks

**Comment by patriksvensson** [comment, marked as answer]
@nathangiuliani Add a reference to the `Cake.NuGet` package in your project and then add the following code to the `Configure` method.

```csharp
public void Configure(ICakeServices services)
{
    // ...

    // Register the NuGet module
    var module = new NuGetModule(new CakeConfiguration(new Dictionary<string, string>()));
    module.Register(services);

   // Register the tools
    services.UseTool(new Uri("nuget:?package=OctopusTools&version=7.4.2"));
    services.UseTool(new Uri("nuget:?package=NUnit.ConsoleRunner&version=3.11.1"));
    services.UseTool(new Uri("nuget:?package=OpenCover&version=4.7.922"));
}
```

We will add support for this automatically in the 1.0 release of Cake (so you just need to add the `UseTool` calls).

**Reply by pascalberger** [reply]
We should also update Tools documentation to fit for Frosting. I've created https://github.com/cake-build/website/issues/1216 for this.

**Reply by patriksvensson** [reply]
@pascalberger Yes, let me fix the issue with NuGet provider not installed by default with Frosting and I'll update the documentation

**Comment by nathangiuliani** [comment]
Thanks @patriksvensson and @pascalberger. All working as expected now - the problem I had originally is, as you expect, the nuget provider not being loaded.
